### ISSUE TITLE
cant diff root node

### ISSUE THREAD

**Issue by max-mapper** [maxogden/dat-core#27]
```
~/Desktop/dat-test 🐈  dat init --no-prompt
Initialized a new dat at /Users/maxogden/Desktop/dat-test
~/Desktop/dat-test 🐈  dat log --json
{"root":true,"change":1,"date":"2015-07-23T02:47:16.050Z","version":"b45458f06ef6fbd1331417e8edd949604298589ca699478cf11a19a63c2f94b7","message":"","links":[],"puts":0,"deletes":0,"files":0}
{"root":false,"change":2,"date":"2015-07-23T02:47:16.061Z","version":"e1badc19d32efb3b0ece9b218578fcd78780590924d45049908a4ba082a9be88","message":"","links":["b45458f06ef6fbd1331417e8edd949604298589ca699478cf11a19a63c2f94b7"],"puts":1,"deletes":0,"files":1}
~/Desktop/dat-test 🐈  dat diff b45458f06ef6fbd1331417e8edd949604298589ca699478cf11a19a63c2f94b7
undefined:1
[object Object]
 ^
SyntaxError: Unexpected token o
    at Object.parse (native)
    at Object.json.decode (/usr/local/lib/node_modules/dat/node_modules/dat-core/lib/encoding.js:6:17)
    at decode (/usr/local/lib/node_modules/dat/node_modules/dat-core/index.js:827:31)
    at DestroyableTransform.filter [as _transform] (/usr/local/lib/node_modules/dat/node_modules/dat-core/index.js:840:12)
    at DestroyableTransform.Transform._read (/usr/local/lib/node_modules/dat/node_modules/through2/node_modules/readable-stream/lib/_stream_transform.js:184:10)
    at DestroyableTransform.Transform._write (/usr/local/lib/node_modules/dat/node_modules/through2/node_modules/readable-stream/lib/_stream_transform.js:172:12)
    at doWrite (/usr/local/lib/node_modules/dat/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:237:10)
    at writeOrBuffer (/usr/local/lib/node_modules/dat/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:227:5)
    at DestroyableTransform.Writable.write (/usr/local/lib/node_modules/dat/node_modules/through2/node_modules/readable-stream/lib/_stream_writable.js:194:11)
    at Class.ondata (/usr/local/lib/node_modules/dat/node_modules/dat-core/node_modules/sorted-diff-stream/node_modules/from2/node_modules/readable-stream/lib/_stream_readable.js:572:20)
~/Desktop/dat-test 🐈  
```

I don't have a use case for diffing the root node, but o/ shouldn't happen

**Comment by mafintosh** [comment]
fixed in 4.0.5 - days since last accident: 0

**Comment by max-mapper** [comment]
lol
